tests/src/main/java/com/hazelcast/stabilizer/tests/map/GrowingMapTest.java

```diff
@@ -69,84 +68,83 @@ public void verify() throws Exception {
     }
 
     private class Worker implements Runnable {
+        private final Random random = new Random();
+
+        private final long[] keys = new long[growCount];
+        private final long[] values = new long[growCount];
+
         @Override
         public void run() {
-            long insertIteration = 0;
-            long deleteIteration = 0;
-            long readIteration = 0;
+            while (!testContext.isStopped()) {
+                int keyIndex = insert();
+                if (readValidation) {
+                    read(keyIndex);
+                }
+                delete(keyIndex);
+            }
+        }
 
-            long[] keys = new long[growCount];
-            long[] values = new long[growCount];
+        private int insert() {
+            int insertIteration = 0;
+            while (!testContext.isStopped() && insertIteration < growCount) {
+                long key = idGenerator.newId();
+                long value = random.nextLong();
 
-            Random random = new Random();
+                keys[insertIteration] = key;
+                values[insertIteration] = value;
 
-            while (!testContext.isStopped()) {
-                int keyIndex = -1;
-                for (int k = 0; k < growCount; k++) {
-                    if (testContext.isStopped()) {
-                        break;
-                    }
+                if (usePut) {
+                    map.put(key, value);
+                } else {
+                    map.set(key, value);
+                }
 
-                    long key = idGenerator.newId();
-                    long value = random.nextLong();
-                    keyIndex = k;
-                    keys[keyIndex] = key;
-                    values[keyIndex] = value;
+                insertIteration++;
+                if (insertIteration % logFrequency == 0) {
+                    log.info(Thread.currentThread().getName() + " At insert iteration: " + insertIteration);
+                }
+            }
 
-                    if (usePut) {
-                        map.put(key, value);
-                    } else {
-                        map.set(key, value);
-                    }
+            log.info(Thread.currentThread().getName() + " Inserted " + insertIteration + " key/value pairs");
+            return insertIteration;
+        }
 
-                    insertIteration++;
-                    if (insertIteration % logFrequency == 0) {
-                        log.info(Thread.currentThread().getName() + " At insert iteration: " + insertIteration);
-                    }
-                }
+        private void read(int keyIndex) {
+            int readIteration = 0;
+            while (!testContext.isStopped() && readIteration < keyIndex) {
```

Comment:
Should be no issue, since there is still (as it was before) an outer `while (!testContext.isStopped())` loop. So the test will repeat the insert, read, delete phases, until it gets stopped. It's a bit weird to read, but should work out.

```
while (!testContext.isStopped()) {
    int howManyKeys = insert();
    if (shouldRead) read(howManyKeys);
    delete(howManyKeys);
}
```